app/models/account.rb

```diff
@@ -15,7 +15,14 @@ class Account < ApplicationRecord
   validates :name, presence: true
 
   scope :visible_for, lambda { |current_user|
-    where(id: [current_user.account_id] + current_user.account.child_ids)
+    where(id: [current_user.account_id] +
+               current_user.account.child_ids +
+               AccountInvitation.accepted_by(current_user).pluck(:account_id))
+  }
+
+  scope :children_and_invitees, lambda { |current_user|
+    where(id: current_user.account.child_ids +
+              AccountInvitation.accepted_by(current_user).pluck(:account_id))
```

Comment:
This is not related to the current task. The current task is only about accept invitation, nothing more.